hpi-swt2/lecture-portal
# Feature/#326 lecturer code

# Resolves #326

This PR changes the feature of user registration.  
I implemented the key as environment variable instead of hard coding it.

Acceptance Criteria:
(In the registration form:)
- [ ] The "is student" property is removed
- [ ] Instead, a field with the title "To become a lecturer, please enter the secret code" is presented
- [ ] The operator of the site is able to hard code a specific code into the application (A String)
- [ ] If a user enters this exact code into the previous mentioned new field, the newly created account will be a Lectures account. If the codes don't match (or the field is left empty) a student account will be created.

Edit account view:
- [ ] In the "Edit Account/Profile"-Section, the isStudent status can not be changed anymore.

Testing steps:
- set the environment variable **SECRET_KEY** to something
- go to the user registration page
- try creating one user with the correct and one with the wrong key, check whether they are lecturer or student

Steps needed for migration:
- None
---

Definition Of Done:
